api/src/opentrons/drivers/thermocycler/driver.py

```
@@ -233,3 +268,43 @@ async def enter_programming_mode(self) -> None:
         await asyncio.sleep(0.05)
         await trigger_connection.close()
         await self._connection.close()
+
+
+class ThermocyclerDriverV2(ThermocyclerDriver):
+    """
+    This driver is for Thermocycler model Gen2.
+    """
+
+    def __init__(self, connection: SerialConnection) -> None:
+        """
+        Constructor
+
+        Args:
+            connection: SerialConnection to the thermocycler
+        """
+        super().__init__(connection)
+
+    async def set_ramp_rate(self, ramp_rate: float) -> None:
+        """Send a set ramp rate command"""
+        # This command is fully unsupported on TC Gen2
+        return None
```

probably want to `raise NotImplementedError()` instead

---

Yeah this one is really weird. My thinking on having it silently return is because that's basically what happens with the Gen1 - the firmware will accept a Set Ramp Rate GCode, but nothing actually happens. Users can specify a ramp rate and it won't throw any exceptions with a Gen1, so that's my only apprehension.